### ISSUE TITLE
UI lets me confirm obviously incorrect SMTP server settings

### ISSUE THREAD

**Issue by danhimalplanet** [mattermost/mattermost-docker#408]
I have a brand new install of Mattermost 5.14.0, which I brought up by using docker-compose.yml with https://github.com/mattermost/mattermost-docker/blob/master/docker-compose.yml

I browed to admin_console/environment/smtp

I entered obviously bogus SMTP server setttings.

I clicked "Test Connection"

The UI confirms the test as valid.
  
<img width="733" alt="Screen Shot 2019-08-17 at 9 16 31 AM" src="https://user-images.githubusercontent.com/1118177/63214702-7d558d00-c0d0-11e9-80d9-1687119e9921.png">

**Comment by triptronics** [comment]
Yeah, I'm still trying to figure out how this passed quality testing.

**Comment by thoschworks** [comment]
> Yeah, I'm still trying to figure out how this passed quality testing. It seems like this is the fix here: [mattermost/mattermost-server#11850](https://github.com/mattermost/mattermost-server/issues/11850)

No that's not a fix, it a crude workaround.

As long as the smtp is not set up, it is not possible to enable the email notification. This results in the situation, that _Test Connection_ may give a wrong positive result.

I think, _Test Connection_ should work correct without enabling the email notifications, because a working smtp setup is the premise for activating the email notifications.

In the present situation everything looks fine but in reality the smtp setup may be broken...

**Comment by aleqx** [comment]
(I just posted in another related issue; doing so here too in case it's monitored by different folks)

I chased my tail for quite a while until I found this thread. Frustrating.

I'd like to make a plea to the devs to **please** give a red **error** instead of "success" when clicking "Test Connection". Something like: "Error: You need to enable Email Notifications in *Site Configuration > Notifications > Enable Email Notifications* in order to activate and test SMTP settings." ... instead of saying "No errors were reported ..." on green. Or actually enable SMTP regardless of the *Email Notifications* setting as @thoschworks sugegsted above.

Saying "No errors" is highly misleading - and time wasting for everyone, including yourselves having to deal with these reports :)
The latter is highly misleading - and time wasting for everyone, including yourselves having to deal with these reports :)

**Comment by agnivade** [comment]
Hello,

Yes, this can indeed be confusing. I have opened 2 issues from the discussion at https://github.com/mattermost/mattermost-server/issues/11850. 

- https://mattermost.atlassian.net/browse/MM-26021
- https://mattermost.atlassian.net/browse/MM-26033

Please feel free to subscribe to them. Thanks.
